packages/block-editor/src/components/spacing-sizes-control/utils.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Checks is given value is a spacing preset.
+ *
+ * @param {string} value Value to check
+ *
+ * @return {boolean} Return true if value is string in format var:preset|spacing|.
+ */
+export function isValueSpacingPreset( value ) {
+	if ( ! value?.includes ) {
+		return false;
+	}
+	return value.includes( 'var:preset|spacing|' );
+}
+
+/**
+ * Converts a spacing preset into a custom value.
+ *
+ * @param {string} value        Value to convert
+ * @param {Array}  spacingSizes Array of the current spacing preset objects
+ *
+ * @return {string} Mapping of the spacing preset to its equivalent custom value.
+ */
+export function getCustomValueFromPreset( value, spacingSizes ) {
+	if ( ! isValueSpacingPreset( value ) ) {
+		return value;
+	}
+
+	const slug = getSpacingPresetSlug( value );
+	const spacingSize = spacingSizes.find( ( size ) => size.slug === slug );
+
+	return spacingSize?.size;
+}
+
+/**
+ * Converts a spacing preset into a custom value.
+ *
+ * @param {string} value Value to convert.
+ *
+ * @return {string} CSS var string for given spacing preset value.
+ */
+export function getSpacingPresetCssVar( value ) {
+	if ( ! value ) {
+		return;
+	}
+	const slug = /var:preset\|spacing\|(.+)/.exec( value );
```

Comment:
Any reason to favour exec over match or even matchAll? Just asking out of interest 😄

Reply:
Copy pasta [from here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/color.js#L277) from memory 😄  - we probably just need match in this instance I think, will look at switching it